### ISSUE TITLE
Dotnet pack does not respect floating version.

### ISSUE THREAD

**Issue by voroninp** [NuGet/Home#8742]
## Details about Problem
I use `nuget pack` and we decided to try floating versions. Alas, it does not work as we expected.

NuGetVersion: NuGet Command Line `5.3.0.4`
dotnet version: `3.0.100`
VS version: 2019 `16.3.6`
OS version: Win10 Pro v1903 (18362.418) (Under parallesl)

## Detailed repro steps so we can see the same problem

1. I have a package with version set to `5.0`

2. This package references another package
```XML
<PackageReference Include="com.luscii.connect.common" Version="*" />
```
3. The latest version of the referenced package is `11.0.97`
4. My expectation is that `nuget pack` scans all available feeds, finds the latest version and sets it accordingly in `nuspec` of created package.
```XML
<dependency id="com.luscii.connect.common" version="[11.0.97,)" exclude="Build,Analyzers" />
```
5. What I see in created package is:
```XML
<dependency id="com.luscii.connect.common" version="5.0.0" exclude="Build,Analyzers" />
```
But there is no version `5.0.0` at all. And why is it fixed?

This is similar to the issue #7232

**Comment by voroninp** [comment]
If I fix major version:
```XML
<PackageReference Include="com.luscii.connect.common" Version="11.*" />
```
, nuspec contains this entry
```XML
<dependency id="com.luscii.connect.common" version="11.0.97" exclude="Build,Analyzers" />
```
This is also not what I need.


Setting version like this:
```XML
<PackageReference Include="com.luscii.connect.common" Version="[11.0.97,)" />
```
Also has no effect on nuspec file.

Meanwhile if I set:
```XML
<PackageReference Include="com.luscii.connect.common" Version="[11.0.97,12)" />
```

I end up having
```XML
<dependency id="com.luscii.connect.common" version="[11.0.97, 12.0.0)" exclude="Build,Analyzers" />
```
But I don't want closed version range. I want to be able to shoot in my leg with autoupgrade of major versions.

**Comment by nkolev92** [comment]
Duplicate of https://github.com/NuGet/Home/issues/6697.

At restore time, NuGet will select the latest version on the feeds.
At pack it will just use whatever was selected at restore time. Pack doesn't go online. 

Note that this bug will be fixed in 5.5
